src/transition.S

```diff
@@ -9,13 +9,14 @@
 .type intercept_wrapper, "function"
 // TODO restore c29
 intercept_wrapper:
+    stp c28, c29, [sp, #-32]!
     mrs c28, ddc
     msr ddc, c29
     stp clr, c28, [sp, #-32]!
     blr x10
     ldp clr, c28, [sp], #32
     msr ddc, c28
-    mov x29, #0
+    ldp c28, c29, [sp], #32
```

Comment:
Since you asked: you can merge the `sp` updates in the `stp` and `ldp` pairs if you reorder things on the stack. This already deviates from the usual calling convention, so that's unlikely to cause issues. To combine the `sp` updates for the `stp`s, you'd have to store the lower (last) two slots first:

```
stp c28, clr, [sp, #-64]!
mrs c28, ddc
msr ddc, c29
stp c28, c29, [sp, #32]
```

The stack then has:
```
[old c29]
[old DDC]
[old clr]
[old c28]  <- sp points here
```
... and I think the operation is otherwise equivalent (but I didn't test it).

To combine the `sp` updates for the `ldp`s, you have to do the reverse, loading the lower address last (so you can post-increment in the last step):

```
ldp c28, c29, [sp, #32]
msr ddc, c28
ldp c28, clr, [sp], #64
```

Reply:
Regarding 

```
stp c28, clr, [sp, #-64]!
mrs c28, ddc
msr ddc, c29
stp c28, c29, [sp, #32]
```

I'm just curious if you swapped the `clr` and `c29` stores for some specific reason? Since we're only writing to (then later reading from) `c28`, that should be the only order-sensitive register in these two code sequences, no?

Reply:
I don't think it's significant. I started off with `stp c29, clr, [...]` (which forms a conventional stack frame) and then worked from there.

Reply:
Alright; in the case, I kept the previous register ordering, but updated the operations to reduce the number of `sp` updates. In https://github.com/capablevms/CHERI-ELF-comp/pull/5/commits/5b1bc067590f88a948e761a17660178f30d0e577.